Sources/ScipioKit/Producer/Cache/CacheSystem.swift

```diff
@@ -269,14 +271,18 @@ struct CacheSystem: Sendable {
         )
     }
 
-    private func retrievePin(product: BuildProduct) throws -> PinsStore.Pin {
-        #if swift(>=5.10)
-        guard let pin = pinsStore.pins[product.package.identity] else {
-            throw Error.revisionNotDetected(product.package.manifest.displayName)
+    private func retrievePin(package: ResolvedPackage) throws -> PinsStore.Pin {
+        #if compiler(>=6.0)
+        guard let pin = pinsStore.pins[package.identity] ?? package.makePinFromRevision() else {
```

Comment:
I'm not sure why the addition is gated by `#if compiler(>=6.0)`. Is that change not compatible with Swift 5.10 compiler?

Reply:
I understand that `GitRepository.getCurrentTag` is added in Swift 6.0 👍

Reply:
Thanks for the question. I should've mentioned that in the PR description. There are two reasons for this: `GitRepository.getCurrentTag` is only available in SwiftPM's `release/6.0` branch. Reimplementing the function would, of course, be trivial, but as I understand it, support for `Xcode 15` is going to be dropped soon, so I thought it wouldn't be worth the effort. Correct me if I'm wrong.

Reply:
Good point. it is correct it will cause the incompatibility.

But I told to @dmhts  that we don't need to maintain the backward compatibility.
https://github.com/giginet/Scipio/pull/140#discussion_r1753282276

So, I think it's no problem if the compatibility is broken because Swift 5 support will be dropped soon. We don't need to make an effort to keep backward compatibility.

I'll drop the Swift 5 support.